### ISSUE TITLE
Internationalization

### ISSUE THREAD

**Issue by thejameskyle** [thejameskyle/babel-plugin-handbook#29]
Internationalization, 国际化, интернационализация, internationalisation, internationalisierung, 国際化, internacionalización, umiędzynarodowienie, internacionalização, 국제화, internazionalizzazione, internationalisering, internacionalizace...

If you understand any of these languages (or any other language) I would be really like if we could work together to translate this handbook.

**Comment by hax** [comment]
`cn.md` should be change to `zh-hans.md` and `zh-hant.md` which stand for simplifeid chinese and tranditional chinese.

**Comment by leandroz** [comment]
I will take care of Spanish, I am from Argentina but I will try to make it as plain Spanish as posible.

**Comment by thejameskyle** [comment]
I'm setting up https://crowdin.com for translations, you'll probably just want to wait for that

**Comment by leandroz** [comment]
Will wait then, post a comment here when is ready!

**Comment by chicoxyzzy** [comment]
> интернационализация

I'll take this one

**Comment by shipengyan** [comment]
I take `中文` `zh-cn.md`

**Comment by kevinwang04** [comment]
I can help zh-cn.md

**Comment by elmosgot** [comment]
I can help with the Dutch translation if needed

**Comment by sashashakun** [comment]
I can help with russian translation.

**Comment by jihchi** [comment]
I can help with Traditional Chinese `正體中文` `zh-tw.md`

**Comment by rmngrc** [comment]
I can help with Spanish :)

**Comment by ryerh** [comment]
国际化, I'm in.

**Comment by leozdgao** [comment]
国际化，I can help with simplified chinese.

**Comment by thejameskyle** [comment]
Hey Everyone! I've setup Crowdin for translations, so if you want to contribute a translation you can do so now. I've updated this issue body above with instructions!

**Comment by thejameskyle** [comment]
:boom: I've setup a digital ocean droplet to update this repo when translations are completed, please let me know if you notice something go wrong

**Comment by thejameskyle** [comment]
Simplified Chinese is now available at: https://github.com/thejameskyle/babel-plugin-handbook/tree/master/translations/zh-CN thanks to @nightire

**Comment by preco21** [comment]
I can help with translate docs into `Korean (ko-KR)`. I'll work soon!

**Comment by nightire** [comment]
@thejameskyle Just FYI, I've fix some broken sentences on Crowdin.

**Comment by Schniz** [comment]
I can help with Hebrew translation

**Comment by elazzabi** [comment]
I'll help with French

**Comment by marcominetti** [comment]
I'll help with Italian!

**Comment by thejameskyle** [comment]
This is so cool, thanks everyone for helping out! Lets keep it going!

<img width="1153" alt="screen shot 2016-01-03 at 1 52 28 am" src="https://cloud.githubusercontent.com/assets/952783/12078211/846f3bca-b1bd-11e5-994c-345e02ab70de.png">

**Comment by andreipfeiffer** [comment]
I will translate to __Romanian__. Happy new year to everyone!

**Comment by andreipfeiffer** [comment]
Please change "Romeno" to "Română", in the translated language list.
Thank you.

**Comment by thejameskyle** [comment]
Whoops, fixed

**Comment by andreipfeiffer** [comment]
Romanian finished.

**Comment by thejameskyle** [comment]
That's awesome! It's times like this I really love open source

**Comment by thejameskyle** [comment]
:hankey: You may have just received a notification that I uploaded new translation strings for a lot of stuff even though the strings haven't changed. I'm looking into it now

**Comment by thejameskyle** [comment]
Would everyone here mind linking me to any language-specific JavaScript communities that we can reach out to? Any forums/newsletters/blogs/social networks?

**Comment by hzoo** [comment]
I remember hearing about iojs translation https://medium.com/@mikeal/how-io-js-built-a-146-person-27-language-localization-effort-in-one-day-65e5b1c49a62#.ovdgke56s

**Comment by thejameskyle** [comment]
That's awesome

**Comment by sotayamashita** [comment]
What about to add base hash to get a difference between original and translated ? such as https://github.com/interagent/http-api-design

**Comment by thejameskyle** [comment]
@sotayamashita I'm sorry, I don't follow

**Comment by nightire** [comment]
well, what @sotayamashita means is like:

```markdown
Portuguese version (based on fba98f08b5), by @Gutem
Spanish version (based on 2a74f45), by @jmnavarro
Korean version (based on f38dba6), by @yoondo
Simplified Chinese version (based on 337c4a0), by @ZhangBohan
Traditional Chinese version (based on 232f8dc), by @kcyeu
Turkish version (based on c03842f), by @hkulekci
```

but it won't be necessary in our case, @thejameskyle setup a automatic tracking & updating system, we, as translators, should do our work on Crowdin.

**Comment by sotayamashita** [comment]
@nightire Oops. I do not understand how Crowdin works.

**Comment by hax** [comment]
Again, `zh-CN` should be renamed to `zh-hans` (because `zh-CN` means Chinese used in mainland China but Singapore also use simplified Chinese), and `zh-TW` should be renamed to `zh-Hant` (because `zh-TW` means Chinese used in Taiwan but Hongkong and Macao also use traditional Chinese).

**Comment by nightire** [comment]
@hax the [original translated version](https://github.com/nightire/babel-plugin-handbook/blob/master/translations%2Fzh-hans.md) is named as `zh-hans` as you suggested. However, since @thejameskyle setup Crowdin as translation platform, he didn't use my version directly,  but I agree with you on naming conversions.

**Comment by hax** [comment]
I found that Crowdin have `zh-HK`, `zh-MO`, `zh-SG` (though titled in "Chinese Traditional, Singapore" which is totally wrong) separately. In practice, I never see any project provide different translations with `zh-HK` vs `zh-MO` (because Macau is too small to have any significant difference with Hong Kong in language usage), or `zh-CN` vs `zh-SG` (because Singapore adopt mainland China standard). So it's the fault of Crowdin, but I expect this project could fix the problem in automation tool.

**Comment by thejameskyle** [comment]
Fixed with https://github.com/thejameskyle/babel-plugin-handbook/commit/aa1a8d48420f1c9015a6fec5377761dd62a197b2 and https://github.com/thejameskyle/babel-plugin-handbook/commit/10d84b892382e6e720a12e17a70cde9e0e78bc4d

**Comment by chicoxyzzy** [comment]
all links in Table of contents are broken now for any translation

**Comment by thejameskyle** [comment]
Yeah, I have no idea how to fix that because they are just the heading titles which are based on the text and crowdin won't let me update the links per translation

**Comment by nightire** [comment]
well, this could be an opportunity to improve crowdin itself :laughing: maybe translators should update links manually on crowdin?

**Comment by thejameskyle** [comment]
It's not currently possible, I can email them about it though

**Comment by d10** [comment]
You can just add `<a id="heading"></a>` to each heading and link to that in the TOC, although I think that wouldn’t work well with GitHub’s autogenerated IDs as there would be two IDs to reference a heading.

**Comment by 0x01f7** [comment]
> You can just add \<a id="heading">\</a> to each heading and link to that in the TOC, although I think that wouldn’t work well with GitHub’s autogenerated IDs as there would be two IDs to reference a heading.

:+1: ... to avoid this, we can use a difference name from heading, or add a prefix to each IDs.

**Comment by pachareal** [comment]
I'll contribute to Russian as well. I've noticed that there are a lot to do for now.

**Comment by ev1stensberg** [comment]
Count me inn on the Norwegian part.

**Comment by jpaulin** [comment]
The single file README.md in Finnish, translated completely. Probably can continue tomorrow with other parts.
